src/components/UserSearch.js

```diff
@@ -0,0 +1,44 @@
+import React, {Component} from 'react';
+import {Route} from 'react-router-dom';
+
+class UserSearch extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      githubUsername: '',
+      selectedUsername: '',
+    };
+
+    this.searchUser = this.searchUser.bind(this);
+    this.updateUsername = this.updateUsername.bind(this);
+  }
+
+  searchUser(evt) {
+    evt.preventDefault();
+    // TODO: Dispatch action that will request data from githubs API
+    console.log("Searching user....");
+    this.setState({selectedUsername: this.state.githubUsername});
+  }
+
+  updateUsername(evt) {
+    console.log(evt);
+    this.setState({
+      githubUsername: evt.target.value,
+    });
+  }
+
+  render() {
+    return (
+      <div>
+        <form onSubmit={this.searchUser}>
+          <label>Github Username</label>
```

Comment:
If using a `label` I would recommend placing the `input`/form field inside the `<label>`. This way if a user clicks on the label text, the `input` would gain focus automatically; as handled by the browser.

Reply:
An alternative would be to use `placeholder` attribute of the `input` tag.

Reply:
Good catch, I can change this.